File: src/main/scala/com/celestia/providers/ServiceProvider.scala
package com.celestia.providers

import com.celestia.interfaces.{ICollisionService, IUpdateService, IRenderService, IServiceProvider}
import com.celestia.services.{CollisionService, UpdateService, RenderService}

/**
 * Created by celestia on 9/24/15.
 */
class ServiceProvider extends IServiceProvider {

  /**
   * Create a new RenderService class
   * @return
   */
  override def getRenderService:IRenderService={
    new RenderService
  }


  /**
   * Create a new UpdateService
   * @return
   */
  override def getUpdateService:IUpdateService={
    new UpdateService
  }


  /**
   * Create a new CollisionService
   * @return
   */
  override def getCollisionService:ICollisionService={
    new CollisionService
  }
}

File: java/src/main/java/com/celestia/csc155/providers/ServiceProvider.java
package com.celestia.csc155.providers;

import com.celestia.csc155.interfaces.ICollisionService;
import com.celestia.csc155.interfaces.IRenderService;
import com.celestia.csc155.interfaces.IServiceProvider;
import com.celestia.csc155.interfaces.IUpdateService;
import com.celestia.csc155.models.GameState;
import com.celestia.csc155.services.CollisionService;
import com.celestia.csc155.services.RenderService;
import com.celestia.csc155.services.UpdateService;

/**
 * Created by celestia on 9/28/15.
 */
public class ServiceProvider implements IServiceProvider {
    
    public IUpdateService getUpdateService() {
        return new UpdateService();
    }

    
    public IRenderService getRenderService() {
        return new RenderService();
    }

    
    public ICollisionService getCollisionService() {
        return new CollisionService();
    }
}
